tools/cell_census_builder/__main__.py

```diff
@@ -210,80 +197,123 @@ def build_step1_get_source_assets(args: argparse.Namespace, assets_path: str) ->
+            dataset_total_cell_count += eb.accumulate_axes(dataset, ad)
+            n += 1
+
+    # populate `var`; create empty `presence` now that we have its dimensions
+    if len(datasets) > 0:
+        for eb in experiment_builders:
+            eb.populate_var_axis()
+
+            # SOMA does not currently support empty arrays, so special case this corner-case.
+            if eb.n_var > 0:
+                max_dataset_joinid = max(d.soma_joinid for d in datasets)
+                eb.experiment.ms["RNA"].add_new_sparse_ndarray(
+                    FEATURE_DATASET_PRESENCE_MATRIX_NAME, type=pa.bool_(), shape=(max_dataset_joinid + 1, eb.n_var)
+                )
+
+    return dataset_total_cell_count
 
-def build_step3_create_X_layers(
+
+def build_step3_populate_axes_and_X_layers(
     assets_path: str,
     filtered_datasets: List[Dataset],
     experiment_builders: List[ExperimentBuilder],
     args: argparse.Namespace,
 ) -> None:
     """
-    Create and populate all X layers
+    Populate all X layers.
     """
     logging.info("Build step 3 - X layer creation - started")
-    # base_path = args.uri
 
-    # Create X layers
-    for e in experiment_builders:
-        e.create_X_layers(filtered_datasets)
-        e.create_joinid_metadata()
+    with ExitStack() as experiments:
+        for eb in experiment_builders:
+            assert eb.experiment.closed
+            # open experiments for write and ensure they are closed when exiting
+            experiments.enter_context(eb.reopen_for_write())
+
+        accumulate_axes(assets_path, filtered_datasets, experiment_builders)
+
+        # Process all X data
+        for eb in experiment_builders:
+            eb.create_X_with_layers()
 
-    # Process all X data
-    populate_X_layers(assets_path, filtered_datasets, experiment_builders, args)
+        populate_X_layers(assets_path, filtered_datasets, experiment_builders, args)
 
-    # tidy up and finish
-    for e in experiment_builders:
-        e.commit_X(consolidate=args.consolidate)
-        e.commit_presence_matrix(filtered_datasets)
+        for eb in experiment_builders:
+            eb.populate_presence_matrix(filtered_datasets)
 
     logging.info("Build step 3 - X layer creation - finished")
 
 
+def build_step4_populate_summary_info(
+    root_collection: soma.Collection,
+    experiment_builders: List[ExperimentBuilder],
+    filtered_datasets: List[Dataset],
+    build_tag: str,
+) -> None:
+    logging.info("Build step 4 - summary info - started")
+
+    with soma.Collection.open(root_collection[CENSUS_INFO_NAME].uri, "w", context=SOMA_TileDB_Context()) as census_info:
+        create_dataset_manifest(census_info, filtered_datasets)
+        create_census_summary_cell_counts(census_info, [e.census_summary_cell_counts for e in experiment_builders])
+        create_census_summary(census_info, experiment_builders, build_tag)
```

Comment:
refactoring these steps into a single method allows for shared use of Collection.open() context manager.

Reply:
Not sure I follow, since all of these methods are scoped within the Collection.open() ctx mgr now.

Reply:
@atolopko-czi - was this follow-up comment directed at some other comment?  I'm having trouble following what you are "not following" :-)

Reply:
OMG, ignore this. 😮‍💨  I did read _my_ "comment to reviewers" as feedback.